src/adapter/graphql/src/queries/datasets/dataset_flow_runs.rs

```diff
@@ -125,11 +117,41 @@ impl DatasetFlowRuns {
             total_count,
         ))
     }
+
+    async fn list_flow_initiators(&self, ctx: &Context<'_>) -> Result<AccountConnection> {
+        utils::check_dataset_read_access(ctx, &self.dataset_handle).await?;
+
+        let flow_service = from_catalog::<dyn fs::FlowService>(ctx).unwrap();
+
+        let flow_initiator_ids: Vec<_> = flow_service
+            .list_all_flow_initiators_by_dataset(&self.dataset_handle.id)
+            .await
+            .int_err()?
+            .matched_stream
+            .try_collect()
+            .await?;
+
+        let matched_flow_initiators: Vec<_> = futures::future::try_join_all(
+            flow_initiator_ids
+                .iter()
+                .map(|initiator_id| Account::from_account_id(ctx, initiator_id.clone())),
+        )
+        .await?;
```

Comment:
Similar problem, maybe less obvious than with datasets. Consider adding a batch version to resolve multiple accounts.